sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala

```diff
@@ -384,9 +384,12 @@ case class CatalogTable(
   def toLinkedHashMap: mutable.LinkedHashMap[String, String] = {
     val map = new mutable.LinkedHashMap[String, String]()
     val tableProperties = properties
-      .filterKeys(!_.startsWith(VIEW_PREFIX))
+      .filter(!_._1.startsWith(VIEW_PREFIX))
+      .filter(kv => !CatalogV2Util.TABLE_RESERVED_PROPERTIES.contains(kv._1))
+      .filter(!_._1.startsWith(TableCatalog.OPTION_PREFIX))
+      .filter(_._1 != TableCatalog.PROP_EXTERNAL)
```

Comment:
hmm, isn't `PROP_EXTERNAL` included in `TABLE_RESERVED_PROPERTIES`?

Reply:
yes.